rampwf/utils/testing.py

```diff
@@ -164,23 +165,24 @@ def assert_submission(ramp_kit_dir='.', ramp_data_dir='.',
                 score_type.name, round(score, score_type.precision)))
 
     print('----------------------------')
-    means = train_train_scoress.mean(axis=0)
-    stds = train_train_scoress.std(axis=0)
-    for mean, std, score_type in zip(means, stds, score_types):
-        print('train {} = {} ± {}'.format(
-            score_type.name, round(mean, score_type.precision),
-            round(std, score_type.precision + 1)))
 
-    means = train_valid_scoress.mean(axis=0)
-    stds = train_valid_scoress.std(axis=0)
-    for mean, std, score_type in zip(means, stds, score_types):
-        print('valid {} = {} ± {}'.format(
-            score_type.name, round(mean, score_type.precision),
-            round(std, score_type.precision + 1)))
+    _print_result(train_train_scoress, score_types, 'train')
+    _print_result(train_valid_scoress, score_types, 'valid')
+    _print_result(test_scoress, score_types, 'test')
+
 
-    means = test_scoress.mean(axis=0)
-    stds = test_scoress.std(axis=0)
+def _print_result(scores, score_types, step):
+    means = scores.mean(axis=0)
+    stds = scores.std(axis=0)
     for mean, std, score_type in zip(means, stds, score_types):
-        print('test {} = {} ± {}'.format(
-            score_type.name, round(mean, score_type.precision),
-            round(std, score_type.precision + 1)))
+        # If std is a NaN
+        if std != std:
```

Comment:
Is this needed? 
It feels a bit a way to hide bugs, as the std can never really be NaN when using actual data?

Reply:
(ah no, it is NaN if your scores are exactly the same, eg all 0 because of using a dummy model)